core/ai_router.py

```
@@ -0,0 +1,319 @@
+            logger.info(f"Falling back to {fallback_provider.value}")
+
+            try:
+                if fallback_provider == AIProvider.OLLAMA:
+                    result = await self._call_ollama(prompt, context)
+                    result["fallback_used"] = True
+                    return result
+                else:
+                    result = await self._call_openai(prompt, context)
+                    result["fallback_used"] = True
+                    return result
+            except Exception as fallback_error:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)